recipes-bsp/u-boot/u-boot-harden.inc

```diff
@@ -39,8 +39,24 @@ SRC_URI:append = "\
     ${TDX_UBOOT_HARDENING_PATCHES} \
 "
 
-do_compile:prepend() {
+do_compile:prepend:imx-generic-bsp() {
     if [ "${TDX_IMX_HAB_ENABLE}" = "0" ] && [ "${TDX_UBOOT_HARDENING_ENABLE}" = "1" ]; then
-        bbfatal 'The combination TDX_IMX_HAB_ENABLE = "0" and TDX_UBOOT_HARDENING_ENABLE = "1" is not allowed: the whitelisting feature (part of the hardening) currently relies on HAB/AHAB.'
+        bbfatal 'The combination TDX_IMX_HAB_ENABLE="0" and TDX_UBOOT_HARDENING_ENABLE="1" is not allowed:' \
+                'the whitelisting feature (part of the hardening) currently relies on HAB/AHAB.'
     fi
 }
+
+k3_compile_sanity_checks() {
+    if [ "${TDX_K3_HSSE_ENABLE}" = "0" ] && [ "${TDX_UBOOT_HARDENING_ENABLE}" = "1" ]; then
+        bbfatal 'The combination TDX_K3_HSSE_ENABLE="0" and TDX_UBOOT_HARDENING_ENABLE="1" is not allowed:' \
+                'the whitelisting feature (part of the hardening) currently relies on K3.'
+    fi
+}
+
+do_compile:prepend:k3() {
+    k3_compile_sanity_checks
+}
+
+do_compile:prepend:k3r5() {
+    k3_compile_sanity_checks
```

Comment:
Yes, I believe so. However, now I realize the check should be different between the two cases (k3, k3r5). In case of k3r5 the hardening cannot be enabled at all at the moment. This might leave some attack surface uncovered but to solve it I'll need to better understand the architecture and what tasks the R5 processor is responsible for.

Reply:
I think U-Boot running on the R5 processor doesn't expose any surface attack (it doesn't have a serial port console, it doesn't use external files for configuration, etc). But I am not 100% sure, so makes sense to do some investigation.

Reply:
Latest version has been updated to have different checks for k3 and k3r5 and as follow up I'm going to create a ticket for further investigations. Nonetheless I've added a note to the documentation saying that we have plans to do that investigation for the sake of transparency:

[9c92d1b](https://github.com/toradex/meta-toradex-security/pull/73/commits/9c92d1ba9ab79a29ef2f822993866a08099933aa) README-secure-boot: update documentation